pkg/scheduler/util/non_zero.go

```diff
@@ -34,45 +37,72 @@ const (
 	DefaultMilliCPURequest int64 = 100 // 0.1 core
 	// DefaultMemoryRequest defines default memory request size.
 	DefaultMemoryRequest int64 = 200 * 1024 * 1024 // 200 MB
+	// DefaultEphemeralStorage defines default empheral storage request size.
+	DefaultEphemeralStorage int64 = 0 // 0 Gi
 )
 
 // GetNonzeroRequests returns the default cpu and memory resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequests(requests *v1.ResourceList) (int64, int64) {
-	return GetNonzeroRequestForResource(v1.ResourceCPU, requests),
-		GetNonzeroRequestForResource(v1.ResourceMemory, requests)
+func GetNonzeroRequests(pod *v1.Pod) (int64, int64) {
+	return GetNonzeroRequestForResource(pod, v1.ResourceCPU),
+		GetNonzeroRequestForResource(pod, v1.ResourceMemory)
 }
 
 // GetNonzeroRequestForResource returns the default resource request if none is found or
 // what is provided on the request.
-func GetNonzeroRequestForResource(resource v1.ResourceName, requests *v1.ResourceList) int64 {
-	switch resource {
-	case v1.ResourceCPU:
-		// Override if un-set, but not if explicitly set to zero
-		if _, found := (*requests)[v1.ResourceCPU]; !found {
-			return DefaultMilliCPURequest
-		}
-		return requests.Cpu().MilliValue()
-	case v1.ResourceMemory:
-		// Override if un-set, but not if explicitly set to zero
-		if _, found := (*requests)[v1.ResourceMemory]; !found {
-			return DefaultMemoryRequest
+func GetNonzeroRequestForResource(pod *v1.Pod, resource v1.ResourceName) int64 {
+	podRequest := resourcehelper.GetResourceRequest(pod, resource)
+	resourceFound := true
+
+	// If Overhead is being utilized, add to the total requests for the pod
+	for i := range pod.Spec.Containers {
+		container := &pod.Spec.Containers[i]
+		requests := container.Resources.Requests
+
+		// If the pod has an explicit resource request(includes initContainers),
+		// we shouldn't assign a default value to it.
+		if podRequest != 0 {
+			break
 		}
-		return requests.Memory().Value()
-	case v1.ResourceEphemeralStorage:
-		quantity, found := (*requests)[v1.ResourceEphemeralStorage]
-		if !found {
-			return 0
+
+		switch resource {
+		case v1.ResourceCPU:
+			// Override if un-set, but not if explicitly set to zero
+			if _, found := (requests)[v1.ResourceCPU]; !found {
+				podRequest = DefaultMilliCPURequest
```

Comment:
directly return the default value (applies elsewhere).

Reply:
I don't think so. As I mentioned in this comment, https://github.com/kubernetes/kubernetes/pull/89222#discussion_r404477824, the `resourcehelper.GetResourceRequest` will return 0, if we just specify the overhead value for CPU and Memroy. In that case, I think we need to consider both of default value and overhead value.

Reply:
If I return the default value directly, the unit test will be failed. https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/types/node_info_test.go#L581

Reply:
Let's wait for @egernst's reply on https://github.com/kubernetes/kubernetes/pull/89222/files#r404517876 first, and then change method `resourcehelper.GetResourceRequest` (or add a new generic method having the extra arg) to fit for both cases.

Reply:
Actually, I discussed this question with sig-api-machinery in here https://kubernetes.slack.com/archives/C0EG7JC6T/p1585995903228600

They wouldn't like to modify the `resourcehelper.GetResourceRequest`

Reply:
True. Then let's add an argument `enforcePodOverhead` to `resourcehelper.GetResourceRequest`; or, if that function is already consumed by a lot of parties, then add a new generic method, and let `resourcehelper.GetResourceRequest` call the new method with `enforcePodOverhead=false`.